File: gamepack/Decompiled/WorldMapArchiveLoader.java
public class WorldMapArchiveLoader {
   boolean isLoaded0 = false;
   int percentLoaded0 = 0;
   String cacheName;
   AbstractArchive archive;

   WorldMapArchiveLoader(AbstractArchive var1) {
      this.archive = var1;
   }

   void reset(String var1) {
      if (var1 != null && !var1.isEmpty()) {
         if (this.cacheName != var1) {
            this.cacheName = var1;
            this.percentLoaded0 = 0;
            this.isLoaded0 = false;
            this.load();
         }
      }
   }

   int load() {
      if (this.cacheName == null) {
         this.percentLoaded0 = 840222292;
         this.isLoaded0 = true;
      } else {
         if (-349750411 * this.percentLoaded0 < 33) {
            if (!this.archive.tryLoadFileByNames(WorldMapCacheName.WorldMapCacheName_compositeMap.name, this.cacheName)) {
               return -349750411 * this.percentLoaded0;
            }

            this.percentLoaded0 = 663820413;
         }

         if (this.percentLoaded0 * -349750411 == 33) {
            if (this.archive.cv(WorldMapCacheName.WorldMapCacheName_compositeTexture.name, this.cacheName) && !this.archive.tryLoadFileByNames(WorldMapCacheName.WorldMapCacheName_compositeTexture.name, this.cacheName)) {
               return this.percentLoaded0 * -349750411;
            }

            this.percentLoaded0 = 1327640826;
         }

         if (this.percentLoaded0 * -349750411 == 66) {
            if (!this.archive.tryLoadFileByNames(this.cacheName, WorldMapCacheName.WorldMapCacheName_labels.name)) {
               return -349750411 * this.percentLoaded0;
            }

            this.percentLoaded0 = 840222292;
            this.isLoaded0 = true;
         }
      }

      return -349750411 * this.percentLoaded0;
   }

   boolean isLoaded() {
      return this.isLoaded0;
   }

   int percentLoaded() {
      return -349750411 * this.percentLoaded0;
   }
}

File: find-names/src/main/java/mapper/identifiers/classes/WorldMapArchiveLoader.kt
package mapper.identifiers.classes

import mapper.abstractclasses.IdentityMapper
import mapper.annotations.DependsOn
import mapper.annotations.MethodParameters
import mapper.predicateutilities.and
import mapper.predicateutilities.predicateOf
import mapper.predicateutilities.type
import mapper.wrappers.ClassWrapper
import mapper.wrappers.FieldWrapper
import mapper.wrappers.MethodWrapper
import org.objectweb.asm.Type.*

@DependsOn(AbstractArchive::class)
class WorldMapArchiveLoader : IdentityMapper.Class() {

    override val predicate = predicateOf<ClassWrapper> { it.superType == Any::class.type }
            .and { it.instanceFields.count { it.type == type<AbstractArchive>() } == 1 }
            .and { it.instanceFields.count { it.type == String::class.type } == 1 }

    class percentLoaded0 : IdentityMapper.InstanceField() {
        override val predicate = predicateOf<FieldWrapper> { it.type == INT_TYPE }
    }

    class cacheName : IdentityMapper.InstanceField() {
        override val predicate = predicateOf<FieldWrapper> { it.type == String::class.type }
    }

    class isLoaded0 : IdentityMapper.InstanceField() {
        override val predicate = predicateOf<FieldWrapper> { it.type == BOOLEAN_TYPE }
    }

    @DependsOn(AbstractArchive::class)
    class archive : IdentityMapper.InstanceField() {
        override val predicate = predicateOf<FieldWrapper> { it.type == type<AbstractArchive>() }
    }

    @MethodParameters()
    class load : IdentityMapper.InstanceMethod() {
        override val predicate = predicateOf<MethodWrapper> { it.returnType == INT_TYPE }
                .and { it.instructions.count() > 10 }
    }

    @MethodParameters()
    class percentLoaded : IdentityMapper.InstanceMethod() {
        override val predicate = predicateOf<MethodWrapper> { it.returnType == INT_TYPE }
                .and { it.instructions.count() < 10 }
    }

    @MethodParameters()
    class isLoaded : IdentityMapper.InstanceMethod() {
        override val predicate = predicateOf<MethodWrapper> { it.returnType == BOOLEAN_TYPE }
    }

    @MethodParameters("cacheName")
    class reset : IdentityMapper.InstanceMethod() {
        override val predicate = predicateOf<MethodWrapper> { it.returnType == VOID_TYPE }
    }
}